Source/Controllers/SubviewControllers/SPTablesList.m

```diff
@@ -1763,6 +1763,31 @@ - (BOOL)selectionShouldChangeInTableView:(nullable NSTableView *)aTableView
 	return [tableDocumentInstance couldCommitCurrentViewActions];
 }
 
+/**
+ * Show warning alert of query execution and return user action result
+ */
+- (BOOL)isQueryExecutionWarningConfirmed:(nonnull NSString *)query
+{
+  if ([prefs boolForKey:SPQueryWarningEnabled] == NO) {
+    return YES;
+  }
+  
+  NSString *message = [NSString stringWithFormat:NSLocalizedString(@"Do you really want to proceed with this query?\n\n %@", @"message of panel asking for confirmation for exec query"), query];
+  
+  NSAlert *alert = [[NSAlert alloc] init];
```

Comment:
Don't we have Alert extension in Swift with helper methods?

Reply:
Would you be able to help guiding me to check if it exists? I don't have any knowledge on Swift. I made these changes based on SPCustomQuery similarities.

Reply:
`[NSAlert createDefaultAlert]` from `NSAlertExtension.swift`

Reply:
Thanks. I just checked! 

Due to the requirement of wrapping up logics into callbacks, it appears to be unsuitable for executing query context. However, it could be a good fit for the "parent" function to obtain user confirmation before invoking the underscore function.